include/deal.II/base/symmetric_tensor.h

```diff
@@ -1217,14 +1218,14 @@ namespace internal
       case 2:
         return (data[0] * sdata[0] +
                 data[1] * sdata[1] +
-                Number(2.) * data[2] * sdata[2]);
+                NumberType<Number>::value(2.0) * data[2] * sdata[2]);
       default:
         // Start with the non-diagonal part to avoid some multiplications by
         // 2.
         Number sum = data[dim] * sdata[dim];
         for (unsigned int d=dim+1; d<(dim*(dim+1)/2); ++d)
           sum += data[d] * sdata[d];
-        sum *= 2.;
+        sum = sum * NumberType<Number>::value(2.0);
```

Comment:
If you do it like this, you can also write `sum *= NumberType<Number>::value(2.0)`, saving 5 characters. ;-)